lld/MachO/Options.td

```diff
@@ -147,6 +147,14 @@ def cs_profile_path: Joined<["--"], "cs-profile-path=">,
 defm pgo_warn_mismatch: BB<"pgo-warn-mismatch",
   "turn on warnings about profile cfg mismatch (default)",
   "turn off warnings about profile cfg mismatch">, Group<grp_lld>;
+def profile_guided_function_order: Joined<["--"], "profile-guided-function-order=">,
+    MetaVarName<"<profile>">, 
+    HelpText<"Read traces from <profile> to order functions to improve startup time">,
+    Group<grp_lld>;
```

Comment:
The name can be confusing given we have `--symbol-ordering-file` and call graph sort using `.cgprofile` section already. 

`--profile-guided-function-order` looks very similar by name, but it's actually very specific for balanced partition layout. 

Also the name suggests the input is an order file, but it's taking an actual profile.

Reply:
I actually thought about splitting this into a "profile" flag (to pass the profile to) and an "enable" flag (just a boolean flag to enable the optimization). Would this be preferred?

As for the flag name, I'm open to suggestions. I could prefix with "bp" for balanced partitioning, but I'm not sure the implementation details are needed in the flag. Like the help text suggests, the point of the flag is to reorder functions based on timestamps from IRPGO profiles. Thus, I thought `--profile-guided-function-order` made sense.

Reply:
`profile-guided` seems too generic, and I felt it also might apply to the call graph profile sort/order.
Given you want to order either function or data for compressions, you might also generalize this option instead of just `function` (although currently it only has function traces).

To match the call graph profile sort flag, what about `-irpgo-profile-sort={irpgo profile file}` or `-irpgo-trace-sort={irpgo profile file}` or something? This flag can imply to order function or data to improve the startup time using IRPGO profiles.
As for ordering for compression, you might also consider consolidating them like `-compression-sort=[function|data|both]` or something.

As mentioned in `SectionPriorities.cpp`, I'd make these flags exclusive to the existing call graph profile sort or order file flags.

Reply:
I like `--irpgo-profile-sort` because it aligns better with `--call-graph-profile-sort`. Thanks for the suggestion! In the future we may order data sections to improve startup, so `function` is not in the flag. When that happens, we may need to add another flag like `-irpgo-profile-sort-type={function,data,both}`. For now I'll leave this out.